atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/LightweightOppTokenTest.java

```diff
@@ -0,0 +1,179 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.io.BaseEncoding;
+import com.google.common.primitives.Longs;
+import com.palantir.conjure.java.lib.Bytes;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Locale;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class LightweightOppTokenTest {
+    @Parameter()
+    public String input;
+
+    @Parameters(name = "{index}: input=\"{0}\"")
+    public static Collection<Object[]> data() {
+        return Stream.concat(
+                        Stream.of(
+                                "0F",
+                                "decafbad",
+                                "0123456789ABCDEF",
+                                "0123456789abcdef",
+                                "0123456789Abcdef",
+                                "0123456789aBcdef",
+                                "0123456789abCdef",
+                                "0123456789abcDef",
+                                "0123456789abcdEf",
+                                "0123456789abcdeF",
+                                Long.toHexString(Long.MIN_VALUE),
+                                Long.toHexString(Long.MAX_VALUE)),
+                        Stream.concat(
+                                IntStream.range(0, 32).mapToObj(i -> {
+                                    byte[] bytes = new byte[i * 2];
+                                    ThreadLocalRandom.current().nextBytes(bytes);
+                                    return BaseEncoding.base16().encode(bytes);
+                                }),
+                                IntStream.range(0, 1000).mapToObj(_i -> BaseEncoding.base16()
+                                        .encode(Longs.toByteArray(
+                                                ThreadLocalRandom.current().nextLong())))))
+                .flatMap(input -> Stream.of(input, input.toLowerCase(Locale.ROOT), input.toUpperCase(Locale.ROOT)))
+                .map(input -> new Object[] {input})
+                .collect(Collectors.toList());
+    }
+
+    @Test
+    public void createFromHex() throws Exception {
```

Comment:
Could we break this up into a couple of tests, which each explicitly state what they're testing for?